extensions/functions_string.yaml

```diff
@@ -163,3 +163,355 @@ scalar_functions:
+            description: Input string.
+          - value: "varchar<L1>"
+            name: "pattern"
+            description: The regular expression pattern.
+        return: i8
+      - args:
+          - options: [ CASE_SENSITIVE, CASE_INSENSITIVE ]
+            required: false
+          - value: "fixedchar<L1>"
+            name: "input"
+            description: Input string.
+          - value: "fixedchar<L1>"
+            name: "pattern"
+            description: The regular expression pattern.
+        return: i8
+  -
+    name: regexp_matches
+    description: >-
+      Returns true if part of the input string matches the regular expression pattern.
+    impls:
+      - args:
+          - options: [ CASE_SENSITIVE, CASE_INSENSITIVE ]
+            required: false
+          - value: "string"
+            name: "input"
+            description: Input string.
+          - value: "string"
+            name: "pattern"
+            description: The regular expression pattern.
+        return: "BOOLEAN"
+      - args:
+          - options: [ CASE_SENSITIVE, CASE_INSENSITIVE ]
+            required: false
+          - value: "varchar<L1>"
+            name: "input"
+            description: Input string.
+          - value: "varchar<L1>"
+            name: "pattern"
+            description: The regular expression pattern.
+        return: "BOOLEAN"
+      - args:
+          - options: [ CASE_SENSITIVE, CASE_INSENSITIVE ]
+            required: false
+          - value: "fixedchar<L1>"
+            name: "input"
+            description: Input string.
+          - value: "fixedchar<L1>"
+            name: "pattern"
+            description: The regular expression pattern.
+        return: "BOOLEAN"
+  - name: replace
+    description: >-
+      Replace any occurrence of the substring with the replacement string
+    impls:
+      - args:
+          - value: "string"
+            name: "input"
+            description: Input string.
+          - value: "string"
+            name: "substring"
+            description: The substring to replace.
+          - value: "string"
+            name: "replacement"
+            description: The replacement string.
+        return: "string"
+      - args:
+          - value: "varchar<L1>"
+            name: "input"
+            description: Input string.
+          - value: "varchar<L1>"
+            name: "substring"
+            description: The substring to replace.
+          - value: "varchar<L1>"
+            name: "replacement"
+            description: The replacement string.
+        return: "varchar<L1>"
+      - args:
+          - value: "fixedchar<L1>"
+            name: "input"
+            description: Input string.
+          - value: "fixedchar<L1>"
+            name: "substring"
+            description: The substring to replace.
+          - value: "fixedchar<L1>"
+            name: "replacement"
+            description: The replacement string.
+        return: "fixedchar<L1>"
```

Comment:
This one in particular is fun for fixedchar. I think it works if `substring` and `replacement` have length `L2` and the input and return value have length `L1`, but for differing `substring` and `replacement` lengths you'd get a data-dependent return type.